### ISSUE TITLE
[qustion] How to programmatically dismiss or close a modal

### ISSUE THREAD

**Issue by rernens** [shlomiassaf/angular2-modal#57]
I am trying to find a way to dismiss programmatically a modal.

I can't find exposed methods (such as close() or dismiss() ) to achieve this objective in the code.

Am I missing something ?

**Comment by shlomiassaf** [comment]
When you open a dialog you get back a `Promise<ModalDialogInstance>`

`ModalDialogInstance` has a `dismiss` and `close` functions.

so:
```ts
modal.alert()
    .title('Hello World')
    .body('Modal')
    .open().then( dialog => {
        dialog.close(true); 
    });
```

**Comment by rernens** [comment]
Thank you for your quick answer. I had missed that. 

I am a newcomer in Angular 2 and Typescript so my following questions might be silly but ... 

**First question** :

Now that I am able to save the ModalInstance for a later use (** thanks to you previous answer **),  ...

```
private _myModal : ModalDialogInstance;

modal.alert()
    .title('Hello World')
    .body('Modal')
    .open().then( dialog => {
        this.myModal = dialog;
        this.cancel();   // it works
    });

public cancel() {
    this._myModal.dismiss();
}
```
... how do I invoke the dismiss function from within the Modal.

I have tried to add a button to the title and to the body that invokes the cancel() function but the function does not get invoked, because, I am assuming, the cancel() function is not available within the modal.

What's the trick then ?

**Second question** :

 In your doc you write  ***« Modal as a component, replace the content by supplying a custom component. »***, how to you then make the modal content a component rather than simple html content.

**Third question** : 

I have tried to use `<ng-router></ng-router>` as the body of the modal to see if the content can be loaded using default ng2 route but it did not work. 

No error but `<ng-router></ng-router>`  is not parsed within the modal.

Is it possible to make the content of the modal to be loaded as the result of ng2 routing and if yes, how?

Thanks for your help.
